### DISCUSSION TITLE
Ideas for gas golfing our contracts

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#767] [Ideas]
I'm creating this discussion to use it as a board for posting ideas related to how we can gas golf our contracts.

As evidenced by this [discussion on the ENS forum](https://discuss.ens.domains/t/rfp-stream-platform-for-ens-service-providers), gas costs are relevant when DAOs assess Sablier.

**Comment by andreivladbrg** [comment]
the most efficient way to reduce gas is by decreasing storage operations (add/remove/change)

one idea might be to find a way to remove `refunded` variable from `Amount` struct:

https://github.com/sablier-labs/v2-core/blob/bf16a469b37b07507f6abb772aae3078abd6ac61/src/types/DataTypes.sol#L42

**Reply by smol-ninja** [reply]
We can also experiment with changes like these: https://github.com/sablier-labs/v2-open-ended/pull/19

**Comment by smol-ninja** [comment]
Posting it here because it might have some good ideas for optimizing our code using Yul: [Crypto streaming PoC in Yul](https://github.com/DRIVENpol/crypto_subscriptions/blob/master/src/Eth_Subscription.sol)

---------
Created [an issue](https://github.com/sablier-labs/v2-core/issues/795) in v2-core to experiment with Yul.
